src/Model/PatientDictContainer.py

```diff
@@ -7,25 +12,60 @@
     num_points
     pixluts
 """
+from src.Model.Singleton import Singleton
+
 
+class PatientDictContainer(metaclass=Singleton):
 
-class PatientDictContainer:
+    def __init__(self):
+        # Initialize base requirements
+        self.path = None                # The path of the loaded directory.
+        self.dataset = None             # Dictionary of PyDicom dataset objects.
+        self.filepaths = None           # Dictionary of filepaths.
 
-    def __init__(self, path, dataset, filepaths, **kwargs):
+        self.additional_data = None     # Any additional values that are required (e.g. rois, raw_dvh, raw_contour, etc)
+
+    def set_base_values(self, path, dataset, filepaths, **kwargs):
         """
-        :param path: The path of the loaded directory.
-        :param dataset: Dictionary where keys are slice number/RT modality and values are PyDicom dataset objects.
-        :param filepaths: Dictionary where keys are slice number/RT modality and values are filepaths.
-        :param kwargs: Any additional values that are required (e.g. rois, raw_dvh, raw_contour, etc)
+        Used to initialize the data on the creation of a new patient.
+        :param path:
+        :param dataset:
+        :param filepaths:
+        :param kwargs:
+        :return:
         """
         self.path = path
         self.dataset = dataset
         self.filepaths = filepaths
         self.additional_data = kwargs
 
-    def add(self, key, value):
+    def clear(self):
         """
-        Adds a new attribute to the keyword arguments.
+        Clears the data in order to prepare for a new patient to be opened.
+        """
+        self.path = None
+        self.dataset = None
+        self.filepaths = None
+        self.additional_data = None
+
+    def is_empty(self):
+        """
+        :return: True if class is empty
+        """
+        if self.path is not None:
+            return False
+        if self.dataset is not None:
+            return False
+        if self.filepaths is not None:
+            return False
+        if self.additional_data is not None:
+            return False
+
+        return True
+
+    def set(self, key, value):
```

Comment:
does this have to be public?
Is it something you are deliberately exposing so any new key / value pair can be provided by any consumer of the class?

Reply:
It is deliberately public so that any class interacting with the object can make changes to the model.